### PR TITLE
Home-work for data-types-overview, numbers

### PR DESCRIPTION
**Pull request by NorbaTamara**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// 1. Declare an array containing different data types.
+let diffTypes = [false, 1, 2, 'three', 'four', 5, {six: 6}];
+
+// 2. Iterate through it and collect amount of each data type to object.
+let collectedTypes = {};
+for(let i = 0; i < diffTypes.length; i++){
+    let type = typeof diffTypes[i];
+    collectedTypes[i] = type;
+};
+console.log(collectedTypes);
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js]: Here you don't actually collect the amount of each data type.
The result should be like:
```js
console.log(collectedTypes);
// {boolean: 1, number: 3, string: 2, object: 1}
```

File: Exercises/numbers/numbers.js [hunk 2]
```diff
@@ -1,3 +1,47 @@
+// NUMBERS
+// 1. Assign by hand same value to 3 variables in binary, octal and hexadecimal format.
+let binar = 0b10010;
+let octa = 0o22;
+let hexa = 0x12; // all number are 18 in decimal
+
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: It's better not to cut words in variable names

File: Exercises/numbers/numbers.js [hunk 3]
```diff
@@ -1,3 +1,47 @@
...
+let initial = 483;
+let inBinary = initial.toString(2);
+console.log(inBinary);
+let inOctal = initial.toString(8);
+console.log(inOctal);
+let inHexa = initial.toString(16);
+console.log(inHexa);
+
+// 4. use Math.random to generate random value from 0 to 100.
+let randomizedVar = Math.round((Math.random()*100));
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: Missing spaces around `*`, unnecessary brackets.

File: Exercises/numbers/numbers.js [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+console.log(randomizedVar);
+
+// ADDITIONAL
+// 1. Make function which will take an array with numbers as a parameter and calculate average number and round to hundredths.
+function avgArr(array) {
+   let result = 0;
+   for(let element of array){
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:30]: missing spaces

File: Exercises/numbers/numbers.js [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+     result += element;
+   }
+   return (result/array.length).toFixed(2);
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:33]: same about spaces

File: Exercises/numbers/numbers.js [hunk 6]
```diff
@@ -1,3 +1,47 @@
...
+};
+
+console.log(avgArr([3,3,6]));
+
+// 2. Generate array to pass to function above with values, and length generated by Math.random.
+
+let randomArr = [];
+let arrLength = Math.round(Math.random()*100);
+for(let i = 0; i < arrLength; i++){
+randomArr.push(Math.random()*100);
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: missing spaces and tabulation

File: Exercises/numbers/numbers.js [hunk 7]
```diff
@@ -1,3 +1,47 @@
...
+// 1. Make function which will take an array with numbers as a parameter and calculate average number and round to hundredths.
+function avgArr(array) {
+   let result = 0;
+   for(let element of array){
+     result += element;
+   }
+   return (result/array.length).toFixed(2);
+};
+
+console.log(avgArr([3,3,6]));
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: spaces

**Review by Mike-Savin — CHANGES_REQUESTED** [review]
Looks good, just see the comments

File: Exercises/data-types-overview/data-types-overview.js (hunk 1 above)
Comment (NorbaTamara) [reply Exercises/data-types-overview/data-types-overview.js]: Changed to

const collectedTypes = {
  string: 0,
  number: 0,
  boolean: 0,
  undefined: 0,
  object: 0,
};
for (let i = 0; i < differentTypes.length; i++) {
  if (typeof differentTypes[i] === 'string') {
    collectedTypes.string++;
  } else if (typeof differentTypes[i] === 'number') {
    collectedTypes.number++;
  } else if (typeof differentTypes[i] === 'boolean') {
    collectedTypes.boolean++;
  } else if (typeof differentTypes[i] === 'undefined') {
    collectedTypes.undefined++;
  } else if (typeof differentTypes[i] === 'object') {
    collectedTypes.object++;
  } else {
    console.log('Unknown type');
  }
}
console.log(collectedTypes);

 :) Please review

File: Exercises/data-types-overview/data-types-overview.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
+// DATA TYPES
+// 1. Declare a variable on each type using two options: plain creation, creation with constructor.
+// 2. Use operator typeof on each of them.
+//number
+let plainNumber = 10;
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: always use `const` if you are not reassigning the variable

File: Exercises/data-types-overview/data-types-overview.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    content: 'js',
+};
+
+let copiedObject = originObject; // objects are referenced to the same place in memory so changing of one will cause changing of other
+
+copiedObject.level = 'basic';
+console.log(originObject);
+console.log(copiedObject); // the same content with originObj
+
+let properlyCopiedObject = {...originObject}; //example how we can avoid duplicating
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js:50]: 👍

File: Exercises/data-types-overview/data-types-overview.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+console.log(action);
+
+
+//ADDITIONAL
+// 1. Declare an array containing different data types.
+let diffTypes = [false, 1, 2, 'three', 'four', 5, {six: 6}];
+
+// 2. Iterate through it and collect amount of each data type to object.
+const collectedTypes = {
+  string: 0,
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: let's do just empty object here, and later on will update it with values

File: Exercises/data-types-overview/data-types-overview.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  number: 0,
+  boolean: 0,
+  undefined: 0,
+  object: 0,
+};
+for (let i = 0; i < differentTypes.length; i++) {
+  if (typeof differentTypes[i] === 'string') {
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: `typeof differentTypes[i]` will contain the string value, which is actually key that we need, so you can add this key to object, if it doesn't exist yet, or increment if it is.

**Review by lizarusi — CHANGES_REQUESTED** [review]

File: Exercises/data-types-overview/data-types-overview.js (hunk 8 above)
Comment (NorbaTamara) [reply Exercises/data-types-overview/data-types-overview.js]: Thank you. Got it and changed in the file.

File: Exercises/data-types-overview/data-types-overview.js (hunk 11 above)
Comment (NorbaTamara) [reply Exercises/data-types-overview/data-types-overview.js]: Changed to:

const typesAmount = {};
for (let element of differentTypes){
    let type = typeof element;
    if (typesAmount[type] === undefined){
       typesAmount[type] = 1;
    } else {
        typesAmount[type]++;
    }
}

**Review by Mike-Savin — APPROVED** [review]
Looks good

**Pull request merged**